DeviceCommunications/DeviceComsProtocolParser.cs

```diff
@@ -0,0 +1,194 @@
+﻿using DeviceCommunications.Interfaces;
+using EEGDataHandling;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DeviceCommunications
+{
+    public class DeviceComsProtocolParser : IDeviceComs, IEEGData
+    {
+        // Set capacity to 30 seconds at 500 samples/second
+        private const int QUEUE_CAPACITY = 500 * 30;
+
+        // use an abstracted device coms stream.
+        IDeviceComsStream _DeviceComsStream;
```

Comment:
General code style / opinion question: I started out using the underscore convention for member variables, but removed them when StyleCop suggested against them. I kind of prefer StyleCop's rule, but I can also change it - do you have strong feelings about using a leading '_'?

Reply:
I don't feel strongly either way, apart from the need to be consistent.